src/Middleware.js

```diff
@@ -0,0 +1,86 @@
+import {RULESET, KEYFRAMES, DECLARATION} from './Enum.js'
+import {test, charat, substr, strlen, sizeof, combine} from './Utility.js'
+import {tokenize} from './Tokenizer.js'
+import {stringify} from './Serializer.js'
+import {prefix} from './Prefixer.js'
+
+/**
+ * @param {function[]} collection
+ * @return {function}
+ */
+export function middleware (collection) {
+	var length = sizeof(collection)
+
+	return function (element, index, children, callback) {
+		var output = ''
+
+		for (var i = 0; i < length; i++)
+			output += collection[i](element, index, children, callback) || ''
+
+		return output
+	}
+}
+
+/**
+ * @param {function} callback
+ * @return {function}
+ */
+export function rulesheet (callback) {
+	return function (element) {
+		if (!element.root)
+			if (element = element.return)
+				callback(element)
+	}
+}
+
+/**
+ * @param {object} element
+ * @param {number} index
+ * @param {object[]} children
+ * @param {function} callback
+ */
+export function prefixer (element, index, children, callback) {
+	if (element.return === element.prefix)
+		switch (element.type) {
+			case DECLARATION: element.prefix = prefix(element.value, element.length)
+				break
+			case KEYFRAMES: element.prefix = prefix(stringify(element, index, element.prefix = null, callback), 10)
```

Comment:
`element.prefix` is a string, is it OK (for engines) to change this property's type to null?

Reply:
Yes, JavaScript engines only special type numbers/doubles.